nevergrad/optimization/base.py

```diff
@@ -165,6 +171,53 @@ def num_tell_not_asked(self) -> int:
         """
         return self._num_tell_not_asked
 
+    def penalty(self, candidate: p.Parameter):
+        def float_penalty(x: tp.Union[bool, float]) -> float:
+            """Unifies penalties as float (bool=False becomes 1)."""
+            return 1 if x is False else 0 if x is True else -x if x < 0 else 0
```

Comment:
please **never** use inline functions, this is causing issues when pickling. Although this is probably not important here, another problem is that it cannot be tested independently. In this case testing may definitely be required.
you can define it outside as `def _float_penalty`

Reply:
Done.